test/ArcxAnalyticsSdk.test.ts

```diff
@@ -33,10 +33,9 @@ describe('(unit) ArcxAnalyticsSdk', () => {
   beforeEach(async () => {
     postRequestStub = sinon.stub(postRequestModule, 'postRequest').resolves(TEST_IDENTITY)
     analyticsSdk = await ArcxAnalyticsSdk.init(TEST_API_KEY, TEST_CONFIG)
+    postRequestStub.resetHistory()
   })
 
-  beforeEach(() => postRequestStub.resetHistory())
```

Comment:
```ts
afterEach(() => postRequestStub.resetHistory())
```

Reply:
this is outdated now, check again. But still, we want to do it in the `beforeEach` because of the line
```
    analyticsSdk = await ArcxAnalyticsSdk.init(TEST_API_KEY, TEST_CONFIG)
```
inside `beforeEach`, which hits our stub

Reply:
I propose the way how we can change this test. With this proposition we probably can remove `resetHistory`, if not it looks good

Reply:
I implemented your suggestion, but still I can't remove `resetHistory`from the `beforeEach` because it's needed at multiple places. Let me know if I missed anything